### PR TITLE
DOC-2979 DOC-2980 DOC-3027 expanded VIEWACTIVITY role

### PR DESCRIPTION
**Pull request by andf-crl**
Addresses: DOC-2979 DOC-2980 DOC-3027 
- Clarified `VIEWACTIVITY` role: now permits access to hostnames and IP addresses on Nodes Overview and Node Reports pages in DB Console
- Scope: v21.2, v22.1 (v22.2, for DOC-2979 will be created from v22.1 this summer)
- Removed reference to `VIEWACTIVITYREDACTED` in `VIEWACTIVITY` description given that `VIEWACTIVITY` includes access to these pages.
- Standardized all `VIEWACTIVITY` descriptions across role/user pages and versions to match (keeping role-specific and user-specific language as appropriate, per)

### CODE REVIEW

**Review by dhartunian — COMMENTED** [review]
> Reviewed 4 of 8 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13591)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andf-crl and @dhartunian)

___
*[v21.2/alter-role.md, line 61 at r1](https://reviewable.io/reviews/cockroachdb/docs/13591#-N-UioGZ2OagB2O5wlSX:-N-UioGZ2OagB2O5wlSY:bimg4jf) ([raw file](https://github.com/cockroachdb/docs/blob/78ac28374cdff6a4e497fef7e848dc8495d6f57c/v21.2/alter-role.md#L61)):*
> ```Markdown
> `SQLLOGIN`/`NOSQLLOGIN` | Allow or disallow a role to log in using the SQL CLI with one of the [client authentication methods](authentication.html#client-authentication). The role option to `NOSQLLOGIN` prevents the role from logging in using the SQL CLI with any authentication method while retaining the ability to log in to DB Console. It is possible to have both `NOSQLLOGIN` and `LOGIN` set for a role and `NOSQLLOGIN` takes precedence on restrictions. <br><br>Without any role options all login behavior is permitted.
> `VALID UNTIL` |  The date and time (in the [`timestamp`](timestamp.html) format) after which the [password](#parameters) is not valid.
> `VIEWACTIVITY`/`NOVIEWACTIVITY` | Allow or disallow a role to see other roles' [queries](show-statements.html) and [sessions](show-sessions.html) using `SHOW STATEMENTS`, `SHOW SESSIONS`, and the [Statements](ui-statements-page.html) and [Transactions](ui-transactions-page.html) pages in the DB Console, as well as node hostnames and IP addresses for all users on the **Nodes Overview** and **Node Reports** pages. With `NOVIEWACTIVITY`, the `SHOW` commands show only the role's own data and the DB Console pages are unavailable.<br><br>By default, the role option is set to `NOVIEWACTIVITY` for all non-admin roles.
> ```

Hostnames and IP addresses are not tied to individual users so I would remove `for all users` in the new addition. Please apply that change to all the files as well.

**Review by andf-crl — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13591)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dhartunian)

___
*[v21.2/alter-role.md, line 61 at r1](https://reviewable.io/reviews/cockroachdb/docs/13591#-N-UioGZ2OagB2O5wlSX:-N-Y_VpLCeGFRXhPL3cw:b-yx0my3) ([raw file](https://github.com/cockroachdb/docs/blob/78ac28374cdff6a4e497fef7e848dc8495d6f57c/v21.2/alter-role.md#L61)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

Hostnames and IP addresses are not tied to individual users so I would remove `for all users` in the new addition. Please apply that change to all the files as well.
</blockquote></details>

Thank you, done x 8! I think I misinterpreted your changelog text on #78045: "... will now display unredacted information containing node hostnames and IP addresses for all users with the `VIEWACTIVITY` role option." I see my mistake now, thank you.

FYI: there exists a near-term (post 22.1 GA, but soon after) project to document all missing DB Console pages, including these (presently unlinked) Nodes pages. These will be linked to those resources once written.

**Review by dhartunian — APPROVED** [review]
> Reviewed 8 of 8 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13591)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andf-crl)

**Review by taroface — CHANGES_REQUESTED** [review]
Suggesting some rephrasing and clarity on the DB Console pages in question. Note that the first comment would apply to all the updated docs in this PR.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13591#-:-N-yF6JJB6Dmzb3PYE0n:ba301td)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andf-crl)

___
*[v21.2/alter-role.md, line 61 at r2](https://reviewable.io/reviews/cockroachdb/docs/13591#-N-xmIQHAvlbEJdPrf8s:-N-xmIQHAvlbEJdPrf8t:b4k47fq) ([raw file](https://github.com/cockroachdb/docs/blob/a26e017cacf84a7300bd8b5ce55f5595bb3b8213/v21.2/alter-role.md#L61)):*
> ```Markdown
> `SQLLOGIN`/`NOSQLLOGIN` | Allow or disallow a role to log in using the SQL CLI with one of the [client authentication methods](authentication.html#client-authentication). The role option to `NOSQLLOGIN` prevents the role from logging in using the SQL CLI with any authentication method while retaining the ability to log in to DB Console. It is possible to have both `NOSQLLOGIN` and `LOGIN` set for a role and `NOSQLLOGIN` takes precedence on restrictions. <br><br>Without any role options all login behavior is permitted.
> `VALID UNTIL` |  The date and time (in the [`timestamp`](timestamp.html) format) after which the [password](#parameters) is not valid.
> `VIEWACTIVITY`/`NOVIEWACTIVITY` | Allow or disallow a role to see other roles' [queries](show-statements.html) and [sessions](show-sessions.html) using `SHOW STATEMENTS`, `SHOW SESSIONS`, and the [Statements](ui-statements-page.html) and [Transactions](ui-transactions-page.html) pages in the DB Console, as well as node hostnames and IP addresses on the **Nodes Overview** and **Node Reports** pages. With `NOVIEWACTIVITY`, the `SHOW` commands show only the role's own data and the DB Console pages are unavailable.<br><br>By default, the role option is set to `NOVIEWACTIVITY` for all non-admin roles.
> ```

(NOTE: This comment goes for all the updated docs)

I'm not actually sure which pages those refer to. @dhartunian do you mean the Node Status endpoints linked on the Advanced Debug page?

If so, @andf-crl I'd rephrase as:

```
as well as node hostnames and IP addresses at the **Node Status** endpoints on the [**Advanced Debug**](ui-debug-pages.html#raw-status-endpoints-json) page.
```

Also consider prepending that with a semicolon instead of a comma, for clarity (I found the sentence a little tricky to parse).
___
*[v21.2/create-user.md, line 67 at r2](https://reviewable.io/reviews/cockroachdb/docs/13591#-N-yDpEn1pdwV-nYaNJx:-N-yDpEn1pdwV-nYaNJy:bs11x2f) ([raw file](https://github.com/cockroachdb/docs/blob/a26e017cacf84a7300bd8b5ce55f5595bb3b8213/v21.2/create-user.md#L67)):*
> ```Markdown
> `SQLLOGIN`/`NOSQLLOGIN` | Allow or disallow a role to log in using the SQL CLI with one of the [client authentication methods](authentication.html#client-authentication). The role option to `NOSQLLOGIN` prevents the role from logging in using the SQL CLI with any authentication method while retaining the ability to log in to DB Console. It is possible to have both `NOSQLLOGIN` and `LOGIN` set for a role and `NOSQLLOGIN` takes precedence on restrictions. <br><br>Without any role options all login behavior is permitted.
> `VALID UNTIL` |  The date and time (in the [`timestamp`](timestamp.html) format) after which the [password](#parameters) is not valid.
> `VIEWACTIVITY`/`NOVIEWACTIVITY` | Allow or disallow the user to see other users' [queries](show-statements.html) and [sessions](show-sessions.html) using `SHOW STATEMENTS`, `SHOW SESSIONS`, and the [Statements](ui-statements-page.html) and [Transactions](ui-transactions-page.html) pages in the DB Console, as well as node hostnames and IP addresses on the **Nodes Overview** and **Node Reports** pages. With `NOVIEWACTIVITY`, the `SHOW` commands show only the user's own data and the DB Console pages are unavailable.<br><br>By default, the user option is set to `NOVIEWACTIVITY` for all non-admin users.
> ```

"the role option is set to `NOVIEWACTIVITY` for all non-`admin` users."

I believe this is called "role options" even if applied to a user and not a role: see https://www.cockroachlabs.com/docs/v21.2/create-user#synopsis



_Code quote:_
```Markdown
the user option is set to `NOVIEWACTIVITY` for all non-admin users.
```
___
*[v22.1/alter-role.md, line 61 at r2](https://reviewable.io/reviews/cockroachdb/docs/13591#-N-yEl_V9kF-QaZE8YD6:-N-yDaQb2LVL-0igEemE:b-axiyz7) ([raw file](https://github.com/cockroachdb/docs/blob/a26e017cacf84a7300bd8b5ce55f5595bb3b8213/v22.1/alter-role.md#L61)):*
> ```Markdown
> `SQLLOGIN`/`NOSQLLOGIN` | Allow or disallow a role to log in using the SQL CLI with one of the [client authentication methods](authentication.html#client-authentication). The role option to `NOSQLLOGIN` prevents the role from logging in using the SQL CLI with any authentication method while retaining the ability to log in to DB Console. It is possible to have both `NOSQLLOGIN` and `LOGIN` set for a role and `NOSQLLOGIN` takes precedence on restrictions. <br><br>Without any role options all login behavior is permitted.
> `VALID UNTIL` |  The date and time (in the [`timestamp`](timestamp.html) format) after which the [password](#parameters) is not valid.
> `VIEWACTIVITY`/`NOVIEWACTIVITY` | Allow or disallow a role to see other roles' [queries](show-statements.html) and [sessions](show-sessions.html) using `SHOW STATEMENTS`, `SHOW SESSIONS`, and the [Statements](ui-statements-page.html) and [Transactions](ui-transactions-page.html) pages in the DB Console, as well as node hostnames and IP addresses on the **Nodes Overview** and **Node Reports** pages. With `NOVIEWACTIVITY`, the `SHOW` commands show only the role's own data and the DB Console pages are unavailable.<br><br>By default, the role option is set to `NOVIEWACTIVITY` for all non-admin roles.
> ```

Nit: non-admin roles -> non-`admin` roles
___
*[v22.1/alter-user.md, line 45 at r2](https://reviewable.io/reviews/cockroachdb/docs/13591#-N-yEsFI98GT4J4y5uRJ:-N-yEsFI98GT4J4y5uRK:bvcb1gy) ([raw file](https://github.com/cockroachdb/docs/blob/a26e017cacf84a7300bd8b5ce55f5595bb3b8213/v22.1/alter-user.md#L45)):*
> ```Markdown
> `CONTROLJOB`/`NOCONTROLJOB` | Allow or disallow the user to [pause](pause-job.html), [resume](resume-job.html), and [cancel](cancel-job.html) jobs. Non-admin users cannot control jobs created by admins. <br><br>By default, the parameter is set to `NOCONTROLJOB` for all non-admin users.
> `CANCELQUERY`/`NOCANCELQUERY` | Allow or disallow the user to cancel [queries](cancel-query.html) and [sessions](cancel-session.html) of other users. Without this privilege, users can only cancel their own queries and sessions. Even with this privilege, non-admins cannot cancel admin queries or sessions. This option should usually be combined with `VIEWACTIVITY` so that the user can view other users' query and session information. <br><br>By default, the parameter is set to `NOCANCELQUERY` for all non-admin users.
> `VIEWACTIVITY`/`NOVIEWACTIVITY` | Allow or disallow the user to see other users' [queries](show-statements.html) and [sessions](show-sessions.html) using `SHOW STATEMENTS`, `SHOW SESSIONS`, and the [Statements](ui-statements-page.html) and [Transactions](ui-transactions-page.html) pages in the DB Console, as well as node hostnames and IP addresses on the **Nodes Overview** and **Node Reports** pages. With `NOVIEWACTIVITY`, the `SHOW` commands show only the user's own data and the DB Console pages are unavailable.<br><br>By default, the user option is set to `NOVIEWACTIVITY` for all non-admin users.
> ```

Same suggested rephrase(s) as earlier
___
*[v22.1/create-role.md, line 77 at r2](https://reviewable.io/reviews/cockroachdb/docs/13591#-N-yEycc16tQ03QbmYTV:-N-yEycc16tQ03QbmYTW:b-axiyz7) ([raw file](https://github.com/cockroachdb/docs/blob/a26e017cacf84a7300bd8b5ce55f5595bb3b8213/v22.1/create-role.md#L77)):*
> ```Markdown
> `SQLLOGIN`/`NOSQLLOGIN` | Allow or disallow a role to log in using the SQL CLI with one of the [client authentication methods](authentication.html#client-authentication). The role option to `NOSQLLOGIN` prevents the role from logging in using the SQL CLI with any authentication method while retaining the ability to log in to DB Console. It is possible to have both `NOSQLLOGIN` and `LOGIN` set for a role and `NOSQLLOGIN` takes precedence on restrictions. <br><br>Without any role options all login behavior is permitted.
> `VALID UNTIL` |  The date and time (in the [`timestamp`](timestamp.html) format) after which the [password](#parameters) is not valid.
> `VIEWACTIVITY`/`NOVIEWACTIVITY` | Allow or disallow a role to see other roles' [queries](show-statements.html) and [sessions](show-sessions.html) using `SHOW STATEMENTS`, `SHOW SESSIONS`, and the [Statements](ui-statements-page.html) and [Transactions](ui-transactions-page.html) pages in the DB Console, as well as node hostnames and IP addresses on the **Nodes Overview** and **Node Reports** pages. With `NOVIEWACTIVITY`, the `SHOW` commands show only the role's own data and the DB Console pages are unavailable.<br><br>By default, the role option is set to `NOVIEWACTIVITY` for all non-admin roles.
> ```

Nit: non-admin roles -> non-`admin` roles
___
*[v22.1/create-user.md, line 67 at r2](https://reviewable.io/reviews/cockroachdb/docs/13591#-N-yF1skCxq31UPc40yA:-N-yF1skCxq31UPc40yB:bvcb1gy) ([raw file](https://github.com/cockroachdb/docs/blob/a26e017cacf84a7300bd8b5ce55f5595bb3b8213/v22.1/create-user.md#L67)):*
> ```Markdown
> `SQLLOGIN`/`NOSQLLOGIN` | Allow or disallow a role to log in using the SQL CLI with one of the [client authentication methods](authentication.html#client-authentication). The role option to `NOSQLLOGIN` prevents the role from logging in using the SQL CLI with any authentication method while retaining the ability to log in to DB Console. It is possible to have both `NOSQLLOGIN` and `LOGIN` set for a role and `NOSQLLOGIN` takes precedence on restrictions. <br><br>Without any role options all login behavior is permitted.
> `VALID UNTIL` |  The date and time (in the [`timestamp`](timestamp.html) format) after which the [password](#parameters) is not valid.
> `VIEWACTIVITY`/`NOVIEWACTIVITY` | Allow or disallow the user to see other users' [queries](show-statements.html) and [sessions](show-sessions.html) using `SHOW STATEMENTS`, `SHOW SESSIONS`, and the [Statements](ui-statements-page.html) and [Transactions](ui-transactions-page.html) pages in the DB Console, as well as node hostnames and IP addresses on the **Nodes Overview** and **Node Reports** pages. With `NOVIEWACTIVITY`, the `SHOW` commands show only the user's own data and the DB Console pages are unavailable.<br><br>By default, the user option is set to `NOVIEWACTIVITY` for all non-admin users.
> ```

Same suggested rephrase(s) as earlier

**Review by dhartunian — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13591)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andf-crl)

___
*[v21.2/alter-role.md, line 61 at r2](https://reviewable.io/reviews/cockroachdb/docs/13591#-N-xmIQHAvlbEJdPrf8s:-N01VyFK6Qg9dMLg8ipF:b-aym7np) ([raw file](https://github.com/cockroachdb/docs/blob/a26e017cacf84a7300bd8b5ce55f5595bb3b8213/v21.2/alter-role.md#L61)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

(NOTE: This comment goes for all the updated docs)

I'm not actually sure which pages those refer to. @dhartunian do you mean the Node Status endpoints linked on the Advanced Debug page?

If so, @andf-crl I'd rephrase as:

```
as well as node hostnames and IP addresses at the **Node Status** endpoints on the [**Advanced Debug**](ui-debug-pages.html#raw-status-endpoints-json) page.
```

Also consider prepending that with a semicolon instead of a comma, for clarity (I found the sentence a little tricky to parse).
</blockquote></details>

To clarify the pages in question:
* Node overview (landing page for DB Console) + node details page (when you click on a node in the table)
* Node reports pages on Advanced Debug. It's a bit confusing because the URL is `/reports/nodes` but the naming is "Node Diagnostics" 
* Node endpoints at `/_status/nodes` that return JSON with node metadata

**Review by andf-crl — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13591)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andf-crl, @dhartunian, and @taroface)

___
*[v21.2/alter-role.md, line 61 at r2](https://reviewable.io/reviews/cockroachdb/docs/13591#-N-xmIQHAvlbEJdPrf8s:-N02NgfC9jfqoixhAnx5:b-6z66nj) ([raw file](https://github.com/cockroachdb/docs/blob/a26e017cacf84a7300bd8b5ce55f5595bb3b8213/v21.2/alter-role.md#L61)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

To clarify the pages in question:
* Node overview (landing page for DB Console) + node details page (when you click on a node in the table)
* Node reports pages on Advanced Debug. It's a bit confusing because the URL is `/reports/nodes` but the naming is "Node Diagnostics" 
* Node endpoints at `/_status/nodes` that return JSON with node metadata
</blockquote></details>

Thank you both.

Frankly, there are many concerns with the extant copy surrounding `VIEWACTIVITY` and `VIEWACTIVITYREDACTED`:

- I am able to access the DB Console as a non-`admin` user with `NOVIEWACTIVITY` despite sentence "With `NOVIEWACTIVITY`, ... the DB Console pages are unavailable." This includes ( statements / queries | transactions | nodes overview | node reports / diagnostics ). Hostnames and IP addresses are indeed redacted, as expected. Suggest: correct this sentence to indicate that access is permitted but host/IP are redacted (have done so, assuming).

- alter-user.html is missing VIEWACTIVITYREDACTED entries, despite this role/user working just fine for me and being present on alter-role.html, its alias. Suggest: add VAR to v21.2 and v22.1 alter-user.html pages. Note: certainly out of scope.

- VA and VAR have identical leading description sentences. Note: certainly out of scope.

- Mentioning of currently-un-documented things. In conferring with Tommy, he recommended not referring to components not already defined in the docs, including:
- the `/_status/nodes` endpoint
- the Advanced Debug page

Re: undocumented things: Certainly this will start to change post-GA with DOC-311 and beyond. I intend to flesh out even the Advanced Debugging page. But that planned work must not begin here in this PR. Instead, I have removed explicit references to DB Console components that are presently undocumented, and instead generalized mention around behavior change: redacted or unredacted. Note that this infers DB-Console-wide behavior as regards node hostname and IP address redaction, which I believe to be the case in my explorations. Please correct me if I am wrong.

Ryan, I have broken up this sentence to enhance parsability.
___
*[v21.2/create-user.md, line 67 at r2](https://reviewable.io/reviews/cockroachdb/docs/13591#-N-yDpEn1pdwV-nYaNJx:-N02INl8BmURttc7JOSA:b-tc7cmh) ([raw file](https://github.com/cockroachdb/docs/blob/a26e017cacf84a7300bd8b5ce55f5595bb3b8213/v21.2/create-user.md#L67)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

"the role option is set to `NOVIEWACTIVITY` for all non-`admin` users."

I believe this is called "role options" even if applied to a user and not a role: see https://www.cockroachlabs.com/docs/v21.2/create-user#synopsis
</blockquote></details>

Thank you! 100%
___
*[v22.1/alter-role.md, line 61 at r2](https://reviewable.io/reviews/cockroachdb/docs/13591#-N-yEl_V9kF-QaZE8YD6:-N02IePv1dVFQsod9v5a:b-ucfknh) ([raw file](https://github.com/cockroachdb/docs/blob/a26e017cacf84a7300bd8b5ce55f5595bb3b8213/v22.1/alter-role.md#L61)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

Nit: non-admin roles -> non-`admin` roles
</blockquote></details>

Thanks!

**Review by andf-crl — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13591)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andf-crl, @dhartunian, and @taroface)

___
*[v21.2/alter-role.md, line 61 at r2](https://reviewable.io/reviews/cockroachdb/docs/13591#-N-xmIQHAvlbEJdPrf8s:-N02i7lh41CDQ-PZUrTI:bpn5ikw) ([raw file](https://github.com/cockroachdb/docs/blob/a26e017cacf84a7300bd8b5ce55f5595bb3b8213/v21.2/alter-role.md#L61)):*
<details><summary><i>Previously, andf-crl wrote…</i></summary><blockquote>

Thank you both.

Frankly, there are many concerns with the extant copy surrounding `VIEWACTIVITY` and `VIEWACTIVITYREDACTED`:

- I am able to access the DB Console as a non-`admin` user with `NOVIEWACTIVITY` despite sentence "With `NOVIEWACTIVITY`, ... the DB Console pages are unavailable." This includes ( statements / queries | transactions | nodes overview | node reports / diagnostics ). Hostnames and IP addresses are indeed redacted, as expected. Suggest: correct this sentence to indicate that access is permitted but host/IP are redacted (have done so, assuming).

- alter-user.html is missing VIEWACTIVITYREDACTED entries, despite this role/user working just fine for me and being present on alter-role.html, its alias. Suggest: add VAR to v21.2 and v22.1 alter-user.html pages. Note: certainly out of scope.

- VA and VAR have identical leading description sentences. Note: certainly out of scope.

- Mentioning of currently-un-documented things. In conferring with Tommy, he recommended not referring to components not already defined in the docs, including:
- the `/_status/nodes` endpoint
- the Advanced Debug page

Re: undocumented things: Certainly this will start to change post-GA with DOC-311 and beyond. I intend to flesh out even the Advanced Debugging page. But that planned work must not begin here in this PR. Instead, I have removed explicit references to DB Console components that are presently undocumented, and instead generalized mention around behavior change: redacted or unredacted. Note that this infers DB-Console-wide behavior as regards node hostname and IP address redaction, which I believe to be the case in my explorations. Please correct me if I am wrong.

Ryan, I have broken up this sentence to enhance parsability.
</blockquote></details>

(Mentions of `Advanced Debug` above should further specify `Advanced Debug > Even More Advanced Debugging`)

**Review by taroface — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13591)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andf-crl, @dhartunian, and @taroface)

___
*[v21.2/alter-role.md, line 61 at r2](https://reviewable.io/reviews/cockroachdb/docs/13591#-N-xmIQHAvlbEJdPrf8s:-N02kepS8GRYxSTk7Vcs:b81095b) ([raw file](https://github.com/cockroachdb/docs/blob/a26e017cacf84a7300bd8b5ce55f5595bb3b8213/v21.2/alter-role.md#L61)):*
<details><summary><i>Previously, andf-crl wrote…</i></summary><blockquote>

(Mentions of `Advanced Debug` above should further specify `Advanced Debug > Even More Advanced Debugging`)
</blockquote></details>

I forgot to mention that `Statements` and `Transactions` should both be bolded per docs style.

One nit on new wording: "redact" may be more precise than "hide" (assuming the information is obscured rather than removed - I could be wrong).

Otherwise this LGTM as long as @dhartunian can verify the accuracy!

**Review by andf-crl — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13591)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andf-crl, @dhartunian, and @taroface)

___
*[v21.2/alter-role.md, line 61 at r2](https://reviewable.io/reviews/cockroachdb/docs/13591#-N-xmIQHAvlbEJdPrf8s:-N02nnod6GjkSGhby_Rs:bwn226i) ([raw file](https://github.com/cockroachdb/docs/blob/a26e017cacf84a7300bd8b5ce55f5595bb3b8213/v21.2/alter-role.md#L61)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

I forgot to mention that `Statements` and `Transactions` should both be bolded per docs style.

One nit on new wording: "redact" may be more precise than "hide" (assuming the information is obscured rather than removed - I could be wrong).

Otherwise this LGTM as long as @dhartunian can verify the accuracy!
</blockquote></details>

Done and Done.

**Review by andf-crl — COMMENTED** [review]
Hi @dhartunian , would you mind a quick once-over of the adjusted copy here per Ryan? Assumption: hostname and IP address redaction are DB Console-wide, and not just for the two previously-listed resources. Thanks!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13591#-:-N0HeKWT9geGbCq8Yhm7:b-d91018)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andf-crl, @dhartunian, and @taroface)

**Review by dhartunian — APPROVED** [review]
:lgtm: yes, if the hostname/IP address *are* visible anywhere else, that would be a bug. I've listed the resources previously because I believed them to be the only ones.

> Reviewed 8 of 8 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13591#-:-N0X3bK62JMuawGdYK92:b-6vqr5q)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andf-crl and @taroface)
